src/midje/parsing/1_to_explicit_form/parse_background.clj

```diff
@@ -58,39 +58,48 @@
 (defn- first-form-is-a-state-changer? [forms]
   (#{"before" "after" "around"} (name (ffirst forms))))
 
-(defn- ^{:testable true } separate-individual-changers
+(defn- process-preq-arrow [forms]
+  (let [arrow-seq     (take-arrow-sequence forms)
+        line-override `(pointer/line-number-known
+                         ~(-> arrow-seq first meta :line))
+        tagged-fake   (-> arrow-seq
+                          prerequisite-to-fake
+                          (fakes/tag-as-background-fake line-override))]
```

Comment:
The fix is to manually override the `:position` value, which is derived later in parsing, here, before we loose the line meta data due to syntax manipulation. This is not the cleanest way to do things because it is an override, but it is similar to how the line information is propagated with normal `provided` data, and the parser is quite involved, so I was unable to find a cleaner way to do it.